GH-41186: [C++][Parquet][Doc] Denote PARQUET:field_id in parquet.rst (#41187)
apache/arrow: docs/source/cpp/parquet.rst

  :func:`ArrowWriterProperties::store_schema` was enabled when writing the file;
  otherwise, it is decoded as an Arrow List.

Serialization details
"""""""""""""""""""""

The Arrow schema is serialized as a :ref:`Arrow IPC <format-ipc>` schema message,
then base64-encoded and stored under the ``ARROW:schema`` metadata key in
the Parquet file metadata.

Field Id
--------

The Parquet format supports an optional integer "field id" which can be assigned
to a field. This is used for example in the
`Apache Iceberg specification <https://github.com/apache/iceberg/blob/main/format/spec.md#column-projection>`__.

On the writer side, If ``PARQUET:field_id`` is present as a metadata key on an Arrow field,
and the corresponding value is a non-negative integer, then it will be used as
the "field id" in the Parquet file.

On the reader side, Arrow will convert these "field id"s to a metadata key named
``PARQUET:field_id`` on the corresponding Arrow field.

> Limitations

Reviewer: Can you take a look at the rendered output (you can preview on GitHub UI) and make sure the heading are structurally consistent? Especially, the "limitations" heading below relates to supported data types.

  :func:`ArrowWriterProperties::store_schema` was enabled when writing the file;
  otherwise, it is decoded as an Arrow List.

> Field Id

Reviewer: This section should be moved below "Serialization details" below, so that the structure of the document makes sense (you can preview it in the GitHub UI).

> Field Id
> ----------

Reviewer: The header underline needs to have the same length
```suggestion
Field Id
--------
```

The Parquet format supports an optional integer "field id" which can be assigned

> to a field. This is used in the `iceberg specification <https://github.com/apache/iceberg/blob/main/format/spec.md#column-projection>` __

Reviewer: ```suggestion
to a field. This is used for example in the
`Apache Iceberg specification <https://github.com/apache/iceberg/blob/main/format/spec.md#column-projection>`__.
```

> On writer side, If ``PARQUET:field_id`` is present as a metadata key on a field, and the corresponding

Reviewer 2: nit: this line seems to be too long.

> value is a nonnegative integer, then it will be used as the "field id" in the parquet

Reviewer 2: ```suggestion
value is a non-negative integer, then it will be used as the "field id" in the parquet
```

> On writer side, If ``PARQUET:field_id`` is present as a metadata key on a field, 
> and the corresponding value is a non-negative integer, then it will be used as 
> the "field id" in the parquet file.

Reviewer: ```suggestion
On the writer side, If ``PARQUET:field_id`` is present as a metadata key on an Arrow field, 
and the corresponding value is a non-negative integer, then it will be used as 
the "field id" in the Parquet file.
```

> On reader side, Arrow will convert these "field id"s to a metadata key named
> ``PARQUET:field_id`` on the appropriate field.

Reviewer: ```suggestion
On the reader side, Arrow will convert these "field id"s to a metadata key named
``PARQUET:field_id`` on the corresponding Arrow field.
```

* EncryptionWithFooterKey and EncryptionWithColumnKey modes.
* Encrypted Footer and Plaintext Footer modes.

> Field Id

Reviewer 2: It seems that `Roundtripping Arrow types` above is a better place for field-id.

----------

> The parquet format supports an optional integer field_id which can be assigned

Reviewer: "Parquet", not "parquet".
```suggestion
The Parquet format supports an optional integer field id which can be assigned
```

> to a field. This is used in the `iceberg specification <https://github.com/apache/iceberg/blob/main/format/spec.md#column-projection>` __

Reviewer 2: I'd suggest not mentioning iceberg to be neutral.

Author: I mention it because it's used there. Any better links could also be ok

Reviewer: I find it useful to give context, so perhaps the wording can be improved to make it clear that it's just an example?

On writer side, If ``PARQUET:field_id`` is present as a metadata key on a field, and the corresponding

> value is a nonnegative integer, then it will be used as the field_id in the parquet

Reviewer: ```suggestion
value is a nonnegative integer, then it will be used as the field id in the parquet
```

Author: Aha, the parquet thrift uses "field_id" here, any word is ok for me though

file.

> On reader side, Arrow will convert these field IDs to a metadata key named

Reviewer: We should settle on a single spelling: "field id" or "field ID". @wjones127 What do you think?
